openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -417,7 +445,16 @@ public List<GenSrcFile> generateBallerinaService(Path openAPI, String serviceNam
         } else {
             openAPIDef.getInfo().setTitle(serviceName);
         }
-
+        // Validate the service generation
+        List<String> complexPaths = GeneratorUtils.getComplexPaths(openAPIDef);
+        if (!complexPaths.isEmpty()) {
+            outStream.println("service generation can not be done due to the given openapi definition contains" +
```

Comment:
This should be a warning right?

Reply:
I'm wondering here to add the warning or error severity since this message is a kind of termination of the code and we don't provide any generated ballerina code. As per `bal` command it doesn't give any severity within those unsupported scenarios.

Reply:
If we terminate the program here, is the process status(I mean the exit code of this process) is failure?

Reply:
There should be an exit code with non-zero for this particular scenario. Currently, 3 could indicate that the YAML file uses features that are not supported by the tool.

Reply:
We have decided to do this with a revamp to the exit codes